### DISCUSSION TITLE
Reorganization of core spec repo to allow for easy referencing by domain/network-specific adaptations

### DISCUSSION THREAD

**Discussion by ravi-prakash-v** [beckn/protocol-specifications#252] [Core Specification]
Currently the core spec contains one single Open API Doc. 

I propose we move each schema as a JSON-LD document and autobuild a fully resolved core.yaml based on updates to the schema via an automated script. 

This way we can reference these schema from domain-specific adaptations of core like DSEP and DHP which auto-generates the domain-specific fully resolved Open API file based on commits done to the respective domain-specific schema. 

We can follow a similar model for domain-specific adaptations that will be referred by network-specific adaptation.

@beckn/core-working-group

**Comment by pramodkvarma** [comment]
Makes sense.
